run.go

```diff
@@ -21,6 +21,7 @@ import (
 	"os"
 	"path/filepath"
 	"sync/atomic"
+	"testing"
```

Comment:
Is it really safe to import `testing` package in non-testing package? I mean, I was wondering if there could be side effects by importing `testing`. Even though this doesn't have side effects, adding a depending package would increase binary size, which sounds not good.

Instead of depending `*testing.B`, creating an interface and using it would be better:

```go
type Benchmark interface {
    StartTimer()
    StopTimer()
}
```

Reply:
It's safe, and it's the only way to get the fancy automatic benchmarking logic that can do things like track allocations, etc.

Reply:
OK but for binary size let's use interface.

Reply:
Oh! I see. I was misunderstanding the idea. I was thinking "but then I have to reimplement all the code", but I don't. I just have to have my test program pass in a thing which satisfies the interface.

Clever!